scripts/performance/perf_load/perf_utils.py

```diff
@@ -130,3 +131,51 @@ def gen_input_output(addr_txos, val):
                 addr_txos[address].append(s_a)
 
     return None, None, None
+
+
+class Singleton(type):
+
+    _instance = None
+
+    def __call__(cls, *args, **kwargs):
+        if cls._instance is None:
+            cls._instance = super().__call__(*args, **kwargs)
+        return cls._instance
+
+
+class PoolRegistry(metaclass=Singleton):  # instantiate it once
+
+    def __init__(self, genesis_path=None, promotion_shift=None):
+        self._genesis_path = genesis_path
+        self._promotion_shift = promotion_shift
+        self._pool_data = None
+        self._current_node = None
+
+        # read genesis to get aliases and dests
+        with open(self._genesis_path, 'r') as f:
+            data = f.read()
+            jsons = [json.loads(x) for x in data.split('\n')]
+            aliases = [_json['txn']['data']['data']['alias'] for _json in jsons]
+            dests = [_json['txn']['data']['dest'] for _json in jsons]
+
+        # put all into list of dicts
+        self._pool_data = [
+            {'node_alias': node_alias,
+             'node_dest': node_dest}
+            for node_alias, node_dest in zip(aliases, dests)
+        ]
+
+    # pick random node from pool and set it as current (for demotion)
+    def get_node(self):
```

Comment:
I would call it `select_new_random_node`

Reply:
Done.